.github/workflows/ci.yml

```diff
@@ -0,0 +1,41 @@
+name: CI
+
+on:
+  push:
+    branches:
+      - "HW1"
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v3
+    - name: Install libs
+      run: |
+        sudo apt-get install libgtest-dev
+        sudo apt-get install cppcheck
+        sudo apt-get install clang-format
+        sudo apt-get install valgrind
+        sudo apt-get install lcov
+        cd homerwork_1
+        mkdir -p build
+    - name: Run CMake
+      run: |
+        cd homerwork_1/build
+        cmake ..
+        cmake --build .
+    - name: Test code
+      run: |
+        cd homerwork_1/
+        cppcheck weather/*.h weather/*.c main.c test/test_weather.cpp
+        clang-format -i weather/*.h weather/*.c main.c  test/test_weather.cpp
```

Comment:
use in another step 
+ reuse github action 
https://github.com/marketplace/actions/clang-format-check